utils/post_processing/restructure_borealis.py

```diff
@@ -0,0 +1,507 @@
+	int_time_array = np.empty(num_records)
+
+	data_dims_array = np.empty((num_records, len(data_record[k]["data_descriptors"])))
+
+	rec_idx = 0
+	for k in data_record:
+		# handle unshared fields
+		int_time_array[rec_idx] = data_record[k]["int_time"]
+		sqn_num_array[rec_idx] = data_record[k]["num_sequences"]
+		data_dims_array[rec_idx] = data_record[k]["data_dimensions"]
+
+		# insert data into buffer
+		record_buffer = data_record[k]["data"]
+		data_pos = rec_idx * data_buffer_offset
+		data_end = data_pos + len(record_buffer)
+		data_buffer[data_pos:data_end] = record_buffer
+
+		# insert sequence timestamps into buffer
+		rec_sqn_ts = data_record[k]["sqn_timestamps"]
+		sqn_pos = rec_idx * sqn_buffer_offset
+		sqn_end = sqn_pos + data_record[k]["num_sequences"]
+		sqn_ts_buffer[sqn_pos:sqn_end] = rec_sqn_ts
+
+		rec_noise = data_record[k]["noise_at_freq"]
+		noise_pos = rec_idx * noise_buffer_offset
+		noise_end = noise_pos + data_record[k]["num_sequences"]
+		noise_buffer[noise_pos:noise_end] = rec_noise
+
+		rec_idx += 1
+
+	# write leftover metadata and data
+	data_dict["int_time"] = int_time_array
+	data_dict["num_sequences"] = sqn_num_array
+
+	data_dict["data"] = data_buffer.reshape(data_shape)
+	data_dict["sqn_timestamps"] = sqn_ts_buffer.reshape(sqn_shape)
+	data_dict["noise_at_freq"] = noise_buffer.reshape(noise_shape)
+
+	data_dict["data_descriptors"] = np.insert(data_dict["data_descriptors"], 0, "num_records")
+	data_dict["data_dimensions"] = data_dims_array
+
+	return data_dict
+
+def bfiq_site_to_array(data_record):
+	"""
+	Restructuring method for bfiq data
+	Args:
+		data_record		a timestamped record loaded from an
+		 				hdf5 Borealis bfiq data file
+	Returns:
+		 data_dict:		A dictionary containing the data from data_record
+		 				reformatted to be stored entirely as arrays, or as
+		 				one entry if the field does not change between records
+	"""
+	data_dict = dict()
+	num_records = len(data_record)
+
+	# write shared fields to dictionary
+	k = list(data_record.keys())[0]
+	for field in shared_bfiq:
+		data_dict[field] = data_record[k][field]
+
+	# find maximum number of sequences
+	max_seqs = find_max_sequences(data_record)
+	dims = data_record[k]["data_dimensions"]
+	num_arrays = dims[0]
+	num_beams = dims[2]
+	num_samps = dims[3]
+
+	data_buffer_offset = num_arrays * num_beams * num_samps * max_seqs
+	data_buffer = np.zeros(num_records * data_buffer_offset, dtype=np.complex64)
+	data_shape = (num_records, num_arrays, max_seqs, num_beams, num_samps)
+
+	sqn_buffer_offset = max_seqs
+	sqn_ts_buffer = np.zeros(num_records * max_seqs)
+	sqn_shape = (num_records, max_seqs)
+
+	noise_buffer_offset = max_seqs
+	noise_buffer = np.zeros(num_records * max_seqs)
+	noise_shape = (num_records, max_seqs)
+
+	sqn_num_array = np.empty(num_records)
+	int_time_array = np.empty(num_records)
+
+	data_dims_array = np.empty((num_records, len(data_record[k]["data_descriptors"])))
+
+	rec_idx = 0
+	for k in data_record:
```

Comment:
Reusing the variable 'k' here. Sometimes this can get confusing, not just for the reader, but also in terms of scope. The 'k' within this for loop is _probably_ the one you want, but do you really know for sure? I suggest a different name for one of the variables.